xmodule/tests/test_library_content.py

```diff
@@ -234,121 +281,147 @@ def test_validation_of_course_libraries(self):
         """
         # When source_library_id is blank, the validation summary should say this block needs to be configured:
         self.lc_block.source_library_id = ""
+        self.lc_block.source_library_version = None
         result = self.lc_block.validate()
         assert not result
-        # Validation fails due to at least one warning/message
         assert result.summary
         assert StudioValidationMessage.NOT_CONFIGURED == result.summary.type
 
         # When source_library_id references a non-existent library, we should get an error:
         self.lc_block.source_library_id = "library-v1:BAD+WOLF"
+        self.lc_block.source_library_version = None
         result = self.lc_block.validate()
         assert not result
-        # Validation fails due to at least one warning/message
         assert result.summary
         assert StudioValidationMessage.ERROR == result.summary.type
         assert 'invalid' in result.summary.text
 
-        # When source_library_id is set but the block needs to be updated, the summary should say so:
+        # When source_library_id is set but the block hasn't been synced, the summary should say so:
         self.lc_block.source_library_id = str(self.library.location.library_key)
+        self.lc_block.source_library_version = None
         result = self.lc_block.validate()
         assert not result
-        # Validation fails due to at least one warning/message
         assert result.summary
         assert StudioValidationMessage.WARNING == result.summary.type
         assert 'out of date' in result.summary.text
 
         # Now if we update the block, all validation should pass:
-        self.lc_block.refresh_children()
+        self._sync_lc_block_from_library()
         assert self.lc_block.validate()
 
+        # But updating the library will cause it to fail again as out-of-date:
+        self._add_problems_to_library()
+        result = self.lc_block.validate()
+        assert not result
+        assert result.summary
+        assert StudioValidationMessage.WARNING == result.summary.type
+        assert 'out of date' in result.summary.text
+
+        # And a regular sync will not fix that:
+        self._sync_lc_block_from_library()
+        result = self.lc_block.validate()
+        assert not result
+        assert result.summary
+        assert StudioValidationMessage.WARNING == result.summary.type
+        assert 'out of date' in result.summary.text
+
+        # But a upgrade_to_latest sync will:
+        self._sync_lc_block_from_library(upgrade_to_latest=True)
+        assert self.lc_block.validate()
+
+    def _assert_has_only_N_matching_problems(self, result, n):
+        assert result.summary
+        assert StudioValidationMessage.WARNING == result.summary.type
+        assert f'only {n} matching problem' in result.summary.text
+
```

Comment:
Some new assertion, to check the cases where a library has been updated and then a non-upgrade sync occurs (i.e., an import happens), and _then_ an upgrade-sync occurs.